src/models/supplemental_accessors.jl

```diff
@@ -23,3 +23,17 @@ function get_series_susceptance(b::PhaseShiftingTransformer)
 end
 
 get_series_admittance(b::ACBranch) = 1 / (get_r(b) + get_x(b) * 1im)
+
+function get_max_active_power(d::T) where {T <: Device}
+    if !hasmethod(get_active_power_limits, Tuple{T})
+        throw(MethodError(get_max_active_power, d))
```

Comment:
Now you've compelled me to test it.  It's not the `MethodError` you're looking for.

```
ERROR: MethodError: no method matching iterate(::TestRenDevice)
Closest candidates are:
  iterate(::Core.SimpleVector) at essentials.jl:603
  iterate(::Core.SimpleVector, ::Any) at essentials.jl:603
  iterate(::ExponentialBackOff) at error.jl:253
```

I think you want something like `throw(MethodError(get_max_active_power, tuple(d)))`.  This still might be confusing to users because it doesn't tell them what they are missing.

If you really want these checks, which do have a minor performance cost, I suggest something like `throw(ArgumentError("Missing required methods for running get_max_active_power on $T"))`.  Or some other exception that accepts a string.

Reply:
Keep in mind that if you remove the checks, Julia will tell the user exactly what they are missing.

Reply:
I changed the implementation.